### PR TITLE
MAX_LIGHT count based on max uniforms supported by shaders

### PR DESCRIPTION
**Pull request by apjagdale**
Adding check for number of uniforms supported in VS and FS and assigning appropriate MAX_LIGHTS count

### CODE REVIEW

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRScene.java [hunk 1]
```diff
@@ -73,6 +73,11 @@
      */
     public GVRScene(GVRContext gvrContext) {
         super(gvrContext, NativeScene.ctor());
+
+        if(MAX_LIGHTS == 0) {
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRScene.java]: There is no guarantee the scene will be created on the GL thread.

**Comment by NolaDonato** [comment]
Looks like I was wrong when I told Abhijit it would be OK. I looked into the initialization code and saw GL calls right before the first call to new GVRScene() and I assumed that code ran on the GL thread. Can you suggest another place where we can put this initialization that is on the GL thread?

**Comment by liaxim** [comment]
- add configureRendering (or whatever feels more appropriately named) to GVRConfigurationManager
- call getActivity().getConfigurationManager().configureRendering() from GVRViewManager.onSurfaceCreated right before setMainScene
- determine the max_lights in configureRendering
- expose getter in GVRConfigurationManager to obtain max_lights
- use gvrContext.getActivity().getConfigurationManager() in GVRScene to get the max lights

I can think of few other things that we could centralize there eventually.

**Comment by liaxim** [comment]
Looks good. Will merge after lunch.

**Pull request merged**